### PR TITLE
Redux store and components

### PR DESCRIPTION
**Pull request by thneves**
- Prepare the directory structure
 - Implement App component
 - Implement BookList, BookForm, and Book component
 - Define Redux actions and reducers
 - Initialize Redux store

### CODE REVIEW

File: src/containers/BooksLists.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import PropTypes from 'prop-types';
+// import { useState } from 'react';
+// import { connect } from 'react-redux';
```
Comment (rukundoeric) [inline src/containers/BooksLists.js]: Kindly don't forget to remove those comments in the next milestone.

File: src/containers/BooksLists.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+};
+
+BooksList.defaultProps = {
+  book: {
+    id: '1',
+    title: 'Fly',
+    author: 'JK',
+    category: 'Mistery',
+  },
+};
```
Comment (rukundoeric) [inline src/containers/BooksLists.js]: -  [MANDATORY] BooksList component ...
    - [x]  Should be connected to redux store. 
    - [x]  Should receive a list of books from props. not just single book.
    - [x]  The default book list should be coming from the store in initial state.
    - [x] The table should iterate through the list of books and display them.

File: src/reducers/books.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import * as actions from '../actions/actionTypes';
+
+const initialState = [];
```
Comment (rukundoeric) [inline src/reducers/books.js]: [MANDATORY] default list of books should be initialized here. like so...
```Javascript
  const initialState = [
      {
          id: '1',
          title: 'Fly',
          author: 'JK',
          category: 'Mistery',
      },
       {
          id: '1',
          title: 'Fly',
          author: 'JK',
          category: 'Mistery',
      },
   ]
```

**Review by rukundoeric — CHANGES_REQUESTED** [review]
# Status: ♻️ CHANGES REQUIRED ♻️
_Great job guys so far 👏 💯 👍. However, there are few issues we need to address before we move on, please check the comments below._

- ### To Highlight
  - Follows gitwork-flow.  💯
  - Codebase looks clean and professional. 💯 

- ### To Improve
  - [x] Check the comments below...
  
_Please make all the required changes before you submit another code review. If you have any problems or concerns, you can reach out to me on Slack: Rukundo Eric 🚀 . I wish you all the very best 🥇_

### _Keep working hard._
### _Happy Coding😃_
-----------------------------------------------------------------------------------------------------------------------------------------
**Regards,
[Rukundo Eric](https://rukundo-eric.herokuapp.com/) ,
[LinkedIn](https://www.linkedin.com/in/rukundo-eric/), 
Slack: Rukundo Eric**

**Review by SpaYco — APPROVED** [review]
Hi @thneves, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**